lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -381,6 +381,61 @@ def update_hvinfo(dev, dev_type):
     # We have a (Disk, link, uri) tuple
     update_hvinfo(disk_entry[0], constants.HOTPLUG_TARGET_DISK)
 
+  # Handle KVM command line argument changes
+  try:
+    idx = kvm_cmd.index("-localtime")
+  except ValueError:
+    pass
+  else:
+    kvm_cmd = kvm_cmd[:idx] + ["-rtc", "base=localtime"] + kvm_cmd[idx+1:]
+
+  try:
+    idx = kvm_cmd.index("-balloon")
+  except ValueError:
+    pass
+  else:
+    balloon_args = kvm_cmd[idx+1].split(",")[1:]
+    balloon_str = "virtio-balloon"
+    if balloon_args:
+      balloon_str += ",%s" % ",".join(balloon_args)
+
+    kvm_cmd = kvm_cmd[:idx] + ["-device", balloon_str] + kvm_cmd[idx+2:]
```

Comment:
OK, here I understand why, but the tracking of indices… ouch.

Reply:
I think I can figure out something more readable, let me try...